diff --git a/build/parse.js b/build/parse.js
index ce1f99bc76e7379db044cd8c00db6193b8d80fa5..ad92ebc6505f4444a6c31c1f7606d0c80abbda61 100644
--- a/build/parse.js
+++ b/build/parse.js
@@ -66,7 +66,7 @@ function isEmptyNode(node) {
  */
 function parse(xml) {
     // prevent the parser from logging non-fatal errors
-    const doc = new xmldom_1.DOMParser({ errorHandler() { } }).parseFromString(xml);
+    const doc = new xmldom_1.DOMParser({ errorHandler() { } }).parseFromString(xml.trimStart(), 'text/xml');
     (0, assert_1.default)(doc.documentElement.nodeName === 'plist', 'malformed document. First element should be <plist>');
     let plist = parsePlistXML(doc.documentElement);
     // the root <plist> node gets interpreted as an Array,
